src/test/java/com/thealgorithms/bitmanipulation/HighestSetBitTest.java

```diff
@@ -0,0 +1,21 @@
+package com.thealgorithms.bitmanipulation;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for Highest Set Bit
+ * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+class HighestSetBitTest {
+
+    @Test
+    void testHighestSetBit() {
+        assertEquals(5, HighestSetBit.findHighestSetBit(37).orElse(-1));
+        assertFalse(HighestSetBit.findHighestSetBit(0).isPresent());
+        assertEquals(0, HighestSetBit.findHighestSetBit(1).orElse(-1));
+        assertThrows(IllegalArgumentException.class, () -> HighestSetBit.findHighestSetBit(-37).orElse(-1));
```

Comment:
In order to _get_ the value from some `Optional` object, you can use the [`get()`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--) method. If the value is not there, a suitable exception will be thrown.

```suggestion
        assertEquals(5, HighestSetBit.findHighestSetBit(37).get());
        assertFalse(HighestSetBit.findHighestSetBit(0).isPresent());
        assertEquals(0, HighestSetBit.findHighestSetBit(1).get());
        assertThrows(IllegalArgumentException.class, () -> HighestSetBit.findHighestSetBit(-37));
```

Reply:
@BamaCharanChhandogi feel free to mark it as _resolve_ (similarly to other pieces of this review).